task/control-plane-agent/src/mgs_common.rs

```diff
@@ -143,14 +153,153 @@ impl MgsCommon {
+    /// and firmware slot.
+    /// For the RoT, that is SpComponent::ROT and slot 0(ImageA) or 1(ImageB)
+    /// or SpComponent::STATE0 and slot 0.
+    pub(crate) fn reset_component_trigger(
+        &mut self,
+        update: &SpUpdate,
+        component: SpComponent,
+    ) -> Result<(), SpError> {
+        if self.reset_component_requested != component {
+            return Err(SpError::ResetComponentTriggerWithoutPrepare);
+        }
+        // If we are not resetting the SP_ITSELF, then we may come back here
+        // to reset something else or to run another prepare/trigger on
+        // the same component.
+        self.reset_component_requested.id.fill(0);
+
+        // Resetting the SP through reset_component() is
+        // the same as through reset() until transient bank selection is
+        // figured out for the SP.
+        match component {
+            SpComponent::SP_ITSELF => {
+                task_jefe_api::Jefe::from(crate::JEFE.get_task_id())
+                    .request_reset();
+                // If `request_reset()` returns,
+                // something has gone very wrong.
+                panic!();
+            }
+            SpComponent::ROT => {
+                // mgs_{gimlet,psc,sidecar}.rs deal with any board specific
+                // reset strategy. Here we take care of common SP and RoT cases.
+            }
+            _ => return Err(SpError::RequestUnsupportedForComponent),
+        };
+        // We're dealing with  RoT targets at this point.
+        match update.sprot_task().reset() {
+            Err(SprotError::RspTimeout) => {
+                // This is the expected error if the reset was successful.
+                // It could be that the RoT is out-to-lunch for some other
+                // reason though.
+                // Things for upper layers to do:
+                //   - Check a boot nonce type thing to see if we are in a
+                //     new session.
+                //   - Check that the expected image is now running.
+                //     (Management plane should do that.)
+                //   - Enable staged updates where we don't automatically
+                //     reset after writing an image.
+                ringbuf_entry!(Log::RotReset {
+                    err: SprotError::RspTimeout
+                });
+                Ok(())
+            }
+            Err(err) => {
+                // Some other error occurred.
+                // Update is all-or-nothing at the moment.
+                // The control plane can try to reset the RoT again or it
+                // can start the update process all over again.  We should
+                // be able to make incremental progress if there is some
+                // bug/condition that is degrading SpRot communications.
+                ringbuf_entry!(Log::RotReset { err });
+                Err(SpError::ComponentOperationFailed(err.into()))
+            }
+            Ok(()) => Ok(()),
```

Comment:
Should we log an error in this case, since we expected `Err(RspTimeout)`?

Reply:
I logged an error here, but I'm wondering if we should actually also return an error. That would require a change in gateway-messages, so I didn't do it yet.